### ISSUE TITLE
broadcom_sta fails to build

### ISSUE THREAD

**Issue by bbigras** [NixOS/nixpkgs#206867]
### Steps To Reproduce
1. `nix-build -A linuxKernel.packages.linux_latest_libre.broadcom_sta`

### Build log
```
/build/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:1792:20: error: initialization of 'int (*)(struct wiphy *, struct net_device *, int,  u8,  bool,  const u8 *, struct key_params *)' {aka 'int (*)(struct wiphy *, struct net_device *, int,  unsigned char,  _Bool,  const unsigned char *, struct key_params *)'} from incompatible pointer type 's32 (*)(struct wiphy *, struct net_device *, u8,  bool,  const u8 *, struct key_params *)' {aka 'int (*)(struct wiphy *, struct net_device *, unsigned char,  _Bool,  const unsigned char *, struct key_params *)'} [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
 1792 |         .add_key = wl_cfg80211_add_key,
      |                    ^~~~~~~~~~~~~~~~~~~
/build/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:1792:20: note: (near initialization for 'wl_cfg80211_ops.add_key')
/build/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:1793:20: error: initialization of 'int (*)(struct wiphy *, struct net_device *, int,  u8,  bool,  const u8 *)' {aka 'int (*)(struct wiphy *, struct net_device *, int,  unsigned char,  _Bool,  const unsigned char *)'} from incompatible pointer type 's32 (*)(struct wiphy *, struct net_device *, u8,  bool,  const u8 *)' {aka 'int (*)(struct wiphy *, struct net_device *, unsigned char,  _Bool,  const unsigned char *)'} [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
 1793 |         .del_key = wl_cfg80211_del_key,
      |                    ^~~~~~~~~~~~~~~~~~~
/build/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:1793:20: note: (near initialization for 'wl_cfg80211_ops.del_key')
/build/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:1794:20: error: initialization of 'int (*)(struct wiphy *, struct net_device *, int,  u8,  bool,  const u8 *, void *, void (*)(void *, struct key_params *))' {aka 'int (*)(struct wiphy *, struct net_device *, int,  unsigned char,  _Bool,  const unsigned char *, void *, void (*)(void *, struct key_params *))'} from incompatible pointer type 's32 (*)(struct wiphy *, struct net_device *, u8,  bool,  const u8 *, void *, void (*)(void *, struct key_params *))' {aka 'int (*)(struct wiphy *, struct net_device *, unsigned char,  _Bool,  const unsigned char *, void *, void (*)(void *, struct key_params *))'} [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
 1794 |         .get_key = wl_cfg80211_get_key,
      |                    ^~~~~~~~~~~~~~~~~~~
/build/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:1794:20: note: (near initialization for 'wl_cfg80211_ops.get_key')
/build/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:1795:28: error: initialization of 'int (*)(struct wiphy *, struct net_device *, int,  u8,  bool,  bool)' {aka 'int (*)(struct wiphy *, struct net_device *, int,  unsigned char,  _Bool,  _Bool)'} from incompatible pointer type 's32 (*)(struct wiphy *, struct net_device *, u8,  bool,  bool)' {aka 'int (*)(struct wiphy *, struct net_device *, unsigned char,  _Bool,  _Bool)'} [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
 1795 |         .set_default_key = wl_cfg80211_config_default_key,
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:1795:28: note: (near initialization for 'wl_cfg80211_ops.set_default_key')
cc1: some warnings being treated as errors
make[2]: *** [/nix/store/vc4xd7fz34rvr5i64j2d0w4gxhgmyqbx-linux-6.1-dev/lib/modules/6.1.0-gnu/source/scripts/Makefile.build:250: /build/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.o] Error 1
make[1]: *** [/nix/store/vc4xd7fz34rvr5i64j2d0w4gxhgmyqbx-linux-6.1-dev/lib/modules/6.1.0-gnu/source/Makefile:1992: /build/broadcom-sta] Error 2
make[1]: Leaving directory '/nix/store/vc4xd7fz34rvr5i64j2d0w4gxhgmyqbx-linux-6.1-dev/lib/modules/6.1.0-gnu/build'
make: *** [Makefile:161: all] Error 2
error: builder for '/nix/store/pi4n8q67lpw4qgirm767cq56z3rk4xbc-broadcom-sta-6.30.223.271-6.1.drv' failed with exit code 2;
       last 10 log lines:
       > /build/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:1794:20: note: (near initialization for 'wl_cfg80211_ops.get_key')
       > /build/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:1795:28: error: initialization of 'int (*)(struct wiphy *, struct net_device *, int,  u8,  bool,  bool)' {aka 'int (*)(struct wiphy *, struct net_device *, int,  unsigned char,  _Bool,  _Bool)'} from incompatible pointer type 's32 (*)(struct wiphy *, struct net_device *, u8,  bool,  bool)' {aka 'int (*)(struct wiphy *, struct net_device *, unsigned char,  _Bool,  _Bool)'} [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
       >  1795 |         .set_default_key = wl_cfg80211_config_default_key,
       >       |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > /build/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:1795:28: note: (near initialization for 'wl_cfg80211_ops.set_default_key')
       > cc1: some warnings being treated as errors
       > make[2]: *** [/nix/store/vc4xd7fz34rvr5i64j2d0w4gxhgmyqbx-linux-6.1-dev/lib/modules/6.1.0-gnu/source/scripts/Makefile.build:250: /build/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.o] Error 1
       > make[1]: *** [/nix/store/vc4xd7fz34rvr5i64j2d0w4gxhgmyqbx-linux-6.1-dev/lib/modules/6.1.0-gnu/source/Makefile:1992: /build/broadcom-sta] Error 2
       > make[1]: Leaving directory '/nix/store/vc4xd7fz34rvr5i64j2d0w4gxhgmyqbx-linux-6.1-dev/lib/modules/6.1.0-gnu/build'
       > make: *** [Makefile:161: all] Error 2
       For full logs, run 'nix log /nix/store/pi4n8q67lpw4qgirm767cq56z3rk4xbc-broadcom-sta-6.30.223.271-6.1.drv'.
```

### Notify maintainers

@X9VoiD maybe

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.81, NixOS, 23.05 (Stoat), 23.05.20221205.6e51c97`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(root): `"nixos-21.11pre299825.d8079260a30"`
 - nixpkgs: `/run/current-system/nixpkgs`

**Comment by X9VoiD** [comment]
There's already a PR that fixes this, #206200 (as linked above), but I'd be glad to hear confirmation that it does fix stuff!

**Comment by bbigras** [comment]
Oh, I thought I checked, but I clearly failed, sorry.

Thank you very much for the PR.
